src/zoo.ts

```diff
@@ -0,0 +1,144 @@
+//  Можливість оповіщення відвідувачів за 15 хвилин до закриття і перед відходом.
+//  "Клієнти": Дані клієнтів зберігаються у Відділу реклами.
+//  Відділ реклами використовує цей список для розсилки новин про зоопарк і рекламні акції.
+//  "Відділ реклами": Відповідає за маркетингові та рекламні заходи.
+//  Використовує список клієнтів для розсилки новин про зоопарк і рекламні акції.
+//  "Виручка": Каса збирає дані про виручку за день. Ці дані передаються в Бухгалтерію.
+//  "Бухгалтерія": Відповідає за фінансове управління зоопарку. Розпоряджається бюджетом, включно з оплатою співробітників,
+//  закупівлею корму для тварин і обслуговуванням зоопарку. Зберігає дані про всіх співробітників, тварин і виплати.
+//  Можливість генерувати фінансові звіти.
+//  "Адміністрація": Відповідає за управління співробітниками і тваринами. Може додавати і видаляти співробітників і тварин.
+//  Створює сповіщення про рекламні акції та інші важливі події в зоопарку.
+//  "Тварини": Включає в себе інформацію про кожну тварину, таку як вид, ім'я, вік, здоров'я та інші характеристики.
+//  "Співробітники": Адміністрація може додавати і видаляти співробітників.
+//  Співробітники можуть мати різні посади та обов'язки, які слід враховувати.
+//  "Бюджет": Бухгалтерія розпоряджається бюджетом і стежить за фінансами зоопарку.
+//  Можливість вести бюджетний облік і надавати фінансові звіти.
+
+enum TicketType {
+  EDULT = 5,
+  CHILD = 3,
+  FAMILY = 15,
+}
+type AnimalHealthType = 'sick' | 'good' | 'exelent';
+class Kassa {
+  //  "Клієнти": Дані клієнтів зберігаються у Відділу реклами. dont really want to add _addDepartment propetry here, should I?
+  private _daylyIncome: number = 0;
+  public get daylyIncome() {
+    return this._daylyIncome;
+  }
+  public close() {
+    const summ = this._daylyIncome;
+    this._daylyIncome = 0;
+    return summ;
+  }
+  public increaseDaylyIncome(ammount: number): number {
+    return (this._daylyIncome += ammount);
+  }
+
+  sendTicket(client: Client, clientList: IClientList[]) {
+    clientList.forEach(list => list.addClient(client));
+  }
+}
+
+// class Ticket {
+//   public ticketType!: TicketType;
+//   private price!: number;
+
+//   constructor(ticketType: TicketType, price: number) {
+//     (this.price = price), (this.ticketType = ticketType);
+//   }
+// }
+
+class Client {
+  private readonly _id!: number;
+  private _firstName!: string;
+  private _lastName!: string;
+  private _contacts!: string;
+  constructor(id: number, firstName: string, lastName: string, contacts: string) {
+    ((this._id = id), (this._firstName = firstName)), (this._lastName = lastName), (this._contacts = contacts);
+  }
+
+  public get firstName(): string {
+    return this._firstName;
+  }
+  public get lastName(): string {
+    return this._lastName;
+  }
+  public get id(): number {
+    return this._id;
+  }
+  public get contacts(): string {
+    return this._contacts;
+  }
+}
+interface IClientList {
```

Comment:
The client list is simply a list of clients and not a separate entity. Additionally, "kassa" adds clients when selling the ticket. And there are no requirements about removing clients. Let's implement base logic.